src/Klarna.Common/Extensions/ShipmentExtensions.cs

```diff
@@ -28,9 +28,16 @@ public static PatchedOrderLine GetOrderLine(this IShipment shipment, ICart cart,
                 if (shippingTaxTotal.Amount > 0)
                 {
                     totalTaxAmount = AmountHelper.GetAmount(shippingTaxTotal.Amount);
-                    taxRate = AmountHelper.GetAmount((shippingTaxTotal.Amount / totals.ShippingTotal.Amount) * 100);
 
-                    total = total + totalTaxAmount;
+                    taxRate = AmountHelper.GetAmount(shippingTaxTotal.Amount * 100 /
+                                                     (market.PricesIncludeTax
+                                                         ? totals.ShippingTotal.Amount - shippingTaxTotal.Amount
```

Comment:
Looks a bit confusing, if prices include tax you have to subtract the tax total first?
Could you perhaps introduce a variable for this and give it a useful name?

Reply:
Updated.